packages/admin/cms-admin/src/dam/DataGrid/FolderDataGrid.tsx

```diff
@@ -0,0 +1,345 @@
+import { isFolder } from "../helpers/isFolder";
+import DamContextMenu from "./DamContextMenu";
+import { useFileUpload } from "./fileUpload/useFileUpload";
+import { damFolderQuery, damItemsListQuery } from "./FolderDataGrid.gql";
+import * as sc from "./FolderDataGrid.sc";
+import { FolderHead } from "./FolderHead";
+import { DamSelectionFooter } from "./footer/SelectionFooter";
+import { DamUploadFooter } from "./footer/UploadFooter";
+import { DamItemLabelColumn } from "./label/DamItemLabelColumn";
+import { useDamSearchHighlighting } from "./useDamSearchHighlighting";
+
+export type DamItemSelectionMap = Map<string, "file" | "folder">;
+
+interface FolderDataGridProps extends DamConfig {
+    id?: string;
+    breadcrumbs?: BreadcrumbItem[];
+    filterApi: IFilterApi<DamFilter>;
+    selectionApi: ISelectionApi;
+}
+
+const FolderDataGrid = ({
+    id,
+    filterApi,
+    breadcrumbs,
+    selectionApi,
+    hideContextMenu,
+    hideArchiveFilter,
+    hideMultiselect,
+    renderDamLabel,
+    ...props
+}: FolderDataGridProps): React.ReactElement => {
+    const intl = useIntl();
+    const apolloClient = useApolloClient();
+
+    const [selectionMap, setSelectionMap] = React.useState<DamItemSelectionMap>(new Map());
+    const [showSelectionFooter, setShowSelectionFooter] = React.useState<boolean>(false);
+    const [showUploadFooter, setShowUploadFooter] = React.useState<boolean>(false);
+    const [uploadTargetFolderName, setUploadTargetFolderName] = React.useState<string>();
+
+    const throttledShowSelectionFooter = useThrottledCallback(() => {
+        hideUploadFooter();
+        setShowSelectionFooter(true);
+    }, 500);
+
+    const hideSelectionFooter = React.useCallback(() => {
+        if (throttledShowSelectionFooter.isPending()) {
+            throttledShowSelectionFooter.cancel();
+        }
+
+        setShowSelectionFooter(false);
+    }, [throttledShowSelectionFooter]);
+
+    const throttledShowUploadFooter = useThrottledCallback(({ folderName }: { folderName?: string }) => {
+        hideSelectionFooter();
+        setShowUploadFooter(true);
+        setUploadTargetFolderName(folderName);
+    }, 500);
+
+    const hideUploadFooter = React.useCallback(() => {
+        if (throttledShowUploadFooter.isPending()) {
+            throttledShowUploadFooter.cancel();
+        }
+
+        if (selectionMap.size > 0) {
+            throttledShowSelectionFooter();
+        }
+
+        setShowUploadFooter(false);
+    }, [selectionMap.size, throttledShowSelectionFooter, throttledShowUploadFooter]);
+
+    React.useEffect(() => {
+        if (selectionMap.size > 0) {
+            throttledShowSelectionFooter();
+        } else {
+            hideSelectionFooter();
+        }
+
+        // useEffect should only be executed if the selection changes
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [selectionMap]);
+
+    const dataGridProps = useDataGridRemote({ pageSize: 20 });
```

Comment:
useDataGridRemote also adds props for filtering and sorting. Does the DAM support the props? If no, we might need to disable sorting and filtering for now

Reply:
At the moment, the DAM does not support filtering or sorting via DataGrid. I disabled these features by setting 

```
sortable: false,
hideSortIcons: true,
disableColumnMenu: true,
```

for each column